### PR TITLE
More changes to Gobblin Launcher API

### PR DESCRIPTION
**Pull request by chavdar**
Changes:
    - Adding system config and log to InstanceDrivers
    - Adding ConfigAccessor to for Gobblin instance configs
    - Adding a launcher for JobLauncherExecutionDriver

### CODE REVIEW

**Comment by chavdar** [comment]
@ibuenros @autumnust Can you review?

File: gobblin-runtime/src/main/java/gobblin/runtime/api/GobblinInstanceLauncher.java [hunk 1]
```diff
@@ -29,4 +33,36 @@
...
+
+    /** Config accessor from a no namespaced typesafe config. */
+    public ConfigAccessor(Config cfg) {
+      Config effectiveCfg = cfg.withFallback(getDefaultConfig());
+      this.startTimeoutMs = effectiveCfg.getLong(START_TIMEOUT_MS);
+      this.shutdownTimeoutMs = effectiveCfg.getLong(SHUTDOWN_TIMEOUT_MS);
+    }
+
+    public static Config getDefaultConfig() {
+      return ConfigFactory.parseResources(RESOURCE_NAME).withFallback(ConfigFactory.load());
```
Comment (ibuenros) [inline gobblin-runtime/src/main/java/gobblin/runtime/api/GobblinInstanceLauncher.java]: you can use `ConfigFactory.parseResources(GobblinInstanceLauncher.class, GobblinInstanceLauncher.class.getSimpleName())` instead

File: gobblin-runtime/src/main/java/gobblin/runtime/instance/StandardGobblinInstanceDriver.java [hunk 2]
```diff
@@ -11,26 +11,60 @@
...
 import gobblin.runtime.api.JobCatalog;
 import gobblin.runtime.api.JobExecutionLauncher;
 import gobblin.runtime.api.JobSpecScheduler;
 import gobblin.runtime.job_catalog.InMemoryJobCatalog;
+import gobblin.runtime.job_exec.JobLauncherExecutionDriver;
 import gobblin.runtime.scheduler.ImmediateJobSpecScheduler;
+import gobblin.runtime.std.DefaultConfigurableImpl;
 
 public class StandardGobblinInstanceDriver extends DefaultGobblinInstanceDriverImpl {
+  private ServiceManager _subservices;
```
Comment (ibuenros) [inline gobblin-runtime/src/main/java/gobblin/runtime/instance/StandardGobblinInstanceDriver.java]: Why is this not part of `DefaultGobblinInstanceDriverImpl` instead?

File: gobblin-runtime/src/main/java/gobblin/runtime/instance/StandardGobblinInstanceDriver.java [hunk 3]
```diff
@@ -11,26 +11,60 @@
...
-    super(jobCatalog, jobScheduler, jobLauncher, log);
+  protected StandardGobblinInstanceDriver(Configurable sysConfig, JobCatalog jobCatalog,
+      JobSpecScheduler jobScheduler, JobExecutionLauncher jobLauncher, Optional<Logger> log) {
+    super(sysConfig, jobCatalog, jobScheduler, jobLauncher, log);
+  }
+
+  @Override
+  protected void startUp() throws Exception {
+    List<Service> componentServices = new ArrayList<>();
+    checkComponentService(getJobCatalog(), componentServices);
```
Comment (ibuenros) [inline gobblin-runtime/src/main/java/gobblin/runtime/instance/StandardGobblinInstanceDriver.java]: should this be done in constructor?

File: gobblin-runtime/src/main/java/gobblin/runtime/api/GobblinInstanceLauncher.java (hunk 1 above)
Comment (chavdar) [inline gobblin-runtime/src/main/java/gobblin/runtime/api/GobblinInstanceLauncher.java]: Fixed.

File: gobblin-runtime/src/main/java/gobblin/runtime/instance/StandardGobblinInstanceDriver.java (hunk 2 above)
Comment (chavdar) [inline gobblin-runtime/src/main/java/gobblin/runtime/instance/StandardGobblinInstanceDriver.java]: DefaultGobblinInstanceDriverImpl does not manage sub-components.

File: gobblin-runtime/src/main/java/gobblin/runtime/instance/StandardGobblinInstanceDriver.java (hunk 3 above)
Comment (chavdar) [inline gobblin-runtime/src/main/java/gobblin/runtime/instance/StandardGobblinInstanceDriver.java]: Fixed.

**Comment by ibuenros** [comment]
LGTM. Can you squash the commits a bit before merging?

**Pull request merged**